HINTS.md

```diff
@@ -0,0 +1,24 @@
+## Advanced
+
+It is typical to call [#to_enum](http://ruby-doc.org/core-2.3.1/Object.html#method-i-to_enum) when defining methods for a generic Enumerable, in case no block is passed.
+
+For example:
+
+```ruby
+module Enumerable
+  def some_method
+    return to_enum(__method__) unless block_given? # __method__ is :some_method here
+    ...
+  end
+end
```

Comment:
Should the example be less 'concrete'?  A location where to find out about this kind of thing may be long term more productive.

The "How to do it" is part of the exercise.